x-pack/plugins/security/server/saved_objects/secure_saved_objects_client_wrapper.test.ts

```diff
@@ -609,22 +609,76 @@ describe('#find', () => {
+      getMockCheckPrivilegesFailure
+    );
+
     const options = Object.freeze({ type: type1, namespaces: ['some-ns'] });
-    await expectForbiddenError(client.find, { options });
-  });
+    const result = await client.find(options);
 
-  test(`throws decorated ForbiddenError when type's an array and unauthorized`, async () => {
-    const options = Object.freeze({ type: [type1, type2], namespaces: ['some-ns'] });
-    await expectForbiddenError(client.find, { options });
+    expect(clientOpts.baseClient.find).not.toHaveBeenCalled();
+    expect(clientOpts.auditLogger.savedObjectsAuthorizationFailure).toHaveBeenCalledTimes(1);
+    expect(clientOpts.auditLogger.savedObjectsAuthorizationFailure).toHaveBeenCalledWith(
+      USERNAME,
+      'find',
+      [type1],
+      options.namespaces,
+      [{ spaceId: 'some-ns', privilege: 'mock-saved_object:foo/find' }],
+      { options }
+    );
+    expect(result).toEqual({ page: 1, per_page: 20, total: 0, saved_objects: [] });
   });
 
-  test(`returns result of baseClient.find when authorized`, async () => {
+  test(`returns result of baseClient.find when fully authorized`, async () => {
     const apiCallReturnValue = { saved_objects: [], foo: 'bar' };
     clientOpts.baseClient.find.mockReturnValue(apiCallReturnValue as any);
 
     const options = Object.freeze({ type: type1, namespaces: ['some-ns'] });
     const result = await expectSuccess(client.find, { options });
+    expect(clientOpts.baseClient.find.mock.calls[0][0]).toEqual({
+      ...options,
+      typesAndNamespacesMap: undefined,
+    });
+    expect(result).toEqual(apiCallReturnValue);
+  });
+
+  test(`returns result of baseClient.find when partially authorized`, async () => {
+    clientOpts.checkSavedObjectsPrivilegesAsCurrentUser.mockResolvedValue({
+      hasAllRequested: false,
+      username: USERNAME,
+      privileges: {
+        kibana: [
+          { resource: 'some-ns', privilege: 'mock-saved_object:foo/find', authorized: true },
+          { resource: 'some-ns', privilege: 'mock-saved_object:bar/find', authorized: true },
+          { resource: 'some-ns', privilege: 'mock-saved_object:baz/find', authorized: false },
+          { resource: 'some-ns', privilege: 'mock-saved_object:qux/find', authorized: false },
+          { resource: 'another-ns', privilege: 'mock-saved_object:foo/find', authorized: true },
+          { resource: 'another-ns', privilege: 'mock-saved_object:bar/find', authorized: false },
+          { resource: 'another-ns', privilege: 'mock-saved_object:baz/find', authorized: true },
+          { resource: 'another-ns', privilege: 'mock-saved_object:qux/find', authorized: false },
+        ],
+      },
+    });
+
+    const apiCallReturnValue = { saved_objects: [], foo: 'bar' };
+    clientOpts.baseClient.find.mockReturnValue(apiCallReturnValue as any);
+
+    const options = Object.freeze({
+      type: ['foo', 'bar', 'baz', 'qux'],
+      namespaces: ['some-ns', 'another-ns'],
```

Comment:
paranoid request (part 2/2): can we add a check against a space that we're fully unauthorized from as well? 
```suggestion
      namespaces: ['some-ns', 'another-ns', 'forbidden-ns'],
```